dhcpv4/nclient4/lease.go

```diff
@@ -65,8 +65,8 @@ func (c *Client) Renew(ctx context.Context, lease *Lease, modifiers ...dhcpv4.Mo
 	}
 	if response.MessageType() == dhcpv4.MessageTypeNak {
 		return &ErrNak{
-			Offer: lease.Offer,
-			Nak:   response,
+			Request: lease.Offer,
```

Comment:
Why is this changing?

Reply:
`ErrNak` was previously used solely with request operations that supply an Offer, but this assumption is too restrictive for general use of a NAK error, thus the change from `Offer` to `Request`. This particular case however should pass in the request (which is a "superset" of the offer) instead of the offer directly. Fixed now.

Reply:
`Offer` has now been removed as per https://github.com/insomniacslk/dhcp/pull/470#discussion_r967865608.